tests/sentry/replays/test_project_replay_recording_segment_index.py

```diff
@@ -54,3 +56,95 @@ def test_index(self):
         assert response.data["data"][0]["segmentId"] == 0
         assert response.data["data"][1]["segmentId"] == 1
         assert response.data["data"][2]["segmentId"] == 2
+
+
+class DownloadSegmentsTestCase(TransactionTestCase):
+    # have to use TransactionTestCase because we're using threadpools
+
+    endpoint = "sentry-api-0-project-replay-recording-segment-index"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(self.user)
+        self.replay_id = uuid.uuid4().hex
+        self.url = reverse(
+            self.endpoint,
+            args=(self.organization.slug, self.project.slug, self.replay_id),
+        )
+
+    def test_index_download_basic_compressed(self):
+        for i in range(0, 3):
+            f = File.objects.create(name=f"rr:{i}", type="replay.recording")
+            f.putfile(BytesIO(zlib.compress(f'{{"test":"hello {i}"}}'.encode())))
+            ReplayRecordingSegment.objects.create(
+                replay_id=self.replay_id,
+                project_id=self.project.id,
+                segment_id=i,
+                file_id=f.id,
+            )
+
+        with self.feature("organizations:session-replay"):
+            response = self.client.get(self.url + "?download=true")
+
+        assert response.status_code == 200
+
+        assert response.get("Content-Type") == "application/json"
+        assert b'[{"test":"hello 0"},{"test":"hello 1"},{"test":"hello 2"}]' == b"".join(
+            response.streaming_content
+        )
+
+    def test_index_download_basic_not_compressed(self):
+        for i in range(0, 3):
+            f = File.objects.create(name=f"rr:{i}", type="replay.recording")
+            f.putfile(BytesIO(f'{{"test":"hello {i}"}}'.encode()))
+            ReplayRecordingSegment.objects.create(
+                replay_id=self.replay_id,
+                project_id=self.project.id,
+                segment_id=i,
+                file_id=f.id,
+            )
+
+        with self.feature("organizations:session-replay"):
+            response = self.client.get(self.url + "?download=true")
```

Comment:
Very nitty.  Little disconnected between endpoint and coverage here.  The endpoint expects `?download` is specified.  It doesn't care about its value.